databricks-sdk-java/src/main/java/com/databricks/sdk/core/UserAgent.java

```diff
@@ -20,8 +39,62 @@ public static void withProduct(String product, String productVersion) {
     UserAgent.productVersion = productVersion;
   }
 
+  // Regular expression copied from https://semver.org/.
+  private static final String semVerCore = "(0|[1-9]\\d*)\\.(0|[1-9]\\d*)\\.(0|[1-9]\\d*)";
+  private static final String semVerPrerelease =
+      "(?:-((?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\\.(?:0|[1-9]\\d*|\\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?";
+  private static final String semVerBuildmetadata = "(?:\\+([0-9a-zA-Z-]+(?:\\.[0-9a-zA-Z-]+)*))?";
+  private static final Pattern regexpSemVer =
+      Pattern.compile("^" + semVerCore + semVerPrerelease + semVerBuildmetadata + "$");
+
+  // Sanitize replaces all non-alphanumeric characters with a hyphen. Use this to
+  // ensure that the user agent value is valid. This is useful when the value is not
+  // ensured to be valid at compile time.
+  //
+  // Example: You want to avoid having '/' and ' ' in the value because it will
+  // make downstream applications fail.
+  //
+  // Note: Semver strings are comprised of alphanumeric characters, hyphens, periods
+  // and plus signs. This function will not remove these characters.
+  // see:
+  // 1. https://semver.org/#spec-item-9
+  // 2. https://semver.org/#spec-item-10
+  private static final Pattern regexpAlphanum = Pattern.compile("^[0-9A-Za-z_\\.\\+-]+$");
+  private static final Pattern regexpAlphanumInverse = Pattern.compile("[^0-9A-Za-z_\\.\\+-]");
+
+  public static String sanitize(String s) {
+    return regexpAlphanumInverse.matcher(s).replaceAll("-");
+  }
+
+  public static boolean matchSemVer(String s) throws IllegalArgumentException {
+    if (regexpSemVer.matcher(s).matches()) {
+      return true;
+    }
+    throw new IllegalArgumentException("Invalid semver string: " + s);
+  }
+
+  public static boolean matchAlphanum(String s) throws IllegalArgumentException {
+    if (regexpAlphanum.matcher(s).matches()) {
+      return true;
+    }
+    throw new IllegalArgumentException("Invalid alphanumeric string: " + s);
+  }
+
+  public static boolean matchAlphanumOrSemVer(String s) throws IllegalArgumentException {
```

Comment:
nit: Might be better to use above 2 methods here and throw their respective errors for clearer error message on what is invalid (alphanumeric or semver).

Reply:
You can't use the two methods above as-is; that would only work if we needed to validate that something was _both_ alphanumeric or semver.